src/m365/spo/commands/group/group-add.ts

```diff
@@ -0,0 +1,113 @@
+import { AxiosRequestConfig } from 'axios';
+import { Logger } from '../../../../cli';
+import {CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  name: string;
+  description?: string;
+  allowMembersEditMembership?: boolean;
+  onlyAllowMembersViewMembership?: boolean;
+  allowRequestToJoinLeave?: boolean;
+  autoAcceptRequestToJoinLeave?: boolean;
+  requestToJoinLeaveEmailSetting?: string;
+}
+
+class SpoGroupAddCommand extends SpoCommand {
+  public get name(): string {
+    return commands.GROUP_ADD;
+  }
+
+  public get description(): string {
+    return 'Creates group in the specified site';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.description = !!args.options.description;
+    telemetryProps.allowMembersEditMembership = !!args.options.allowMembersEditMembership;
+    telemetryProps.onlyAllowMembersViewMembership = !!args.options.onlyAllowMembersViewMembership;
+    telemetryProps.allowRequestToJoinLeave = !!args.options.allowRequestToJoinLeave;
+    telemetryProps.autoAcceptRequestToJoinLeave = !!args.options.autoAcceptRequestToJoinLeave;
+    telemetryProps.requestToJoinLeaveEmailSetting = !!args.options.requestToJoinLeaveEmailSetting;
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    (async () => {
+      try {
+        const requestOptions: AxiosRequestConfig = {
+          url: `${args.options.webUrl}/_api/web/sitegroups`,
+          headers: {
+            accept: 'application/json;odata=nometadata'
+          },
+          responseType: 'json',
+          data: {
+            Title: args.options.name,
+            Description: args.options.description,
+            AllowMembersEditMembership: args.options.allowMembersEditMembership,
+            OnlyAllowMembersViewMembership: args.options.onlyAllowMembersViewMembership,
+            AllowRequestToJoinLeave: args.options.allowRequestToJoinLeave,
+            AutoAcceptRequestToJoinLeave: args.options.autoAcceptRequestToJoinLeave,
+            RequestToJoinLeaveEmailSetting: args.options.requestToJoinLeaveEmailSetting
+          }
+        };
+  
+        const response = await request.post(requestOptions);
+  
+        logger.log(response);
+        cb();
+      }
+      catch (err: any) {
+        this.handleRejectedODataJsonPromise(err, logger, cb);
+      }
+    })();
```

Comment:
So we discussed this a bit internally and the approach is totally fine but for the time being I think we should refactor this to work with promises in order to be consistent with what we have now in other places. Keeping similar approach in each command will help us maintain the code.
We may think of a more global refactor of this later on.
Besides that this is ready to be merged 👍.
I think you also have one other PR with similar comment, which is also ready, just waiting for this refactor.

Reply:
Is it fine if I just refactor the `commandAction` method and leave the private methods of the command `async`?

Reply:
IMHO it will be just fine, but I guess it currently more common that the private methods also return promise which we resolve/handle in command action. If we currently keep this approach it would be perfect.👍

Reply:
All right, I personally regret that we can't use async/await. Hopefully this global refactor happens soon so we can use it. ☺️